src/core/js/app.js

```diff
@@ -111,6 +112,9 @@ require([
     }
 
     function mapAdaptIdsToObjects () {
+        // Clear any existing mappings, as they are no longer valid.
+        Adapt.mappedIds = {};
```

Comment:
mapAdaptIdsToObjects  and mappedIds are unrelated. this should go into Adapt.setupMapping

mapAdaptIdsToObjects : is for indexing items (components, blocks, articles) in their collections
mappedIds: is for indexing the collections